tests/lib/Output/Generator/FieldTypeHashGeneratorBaseTest.php

```diff
@@ -283,18 +283,31 @@ private function getGeneratorOutput()
         return $this->getGenerator()->endDocument('Version');
     }
 
-    private function assertSerializationSame($functionName)
+    private function assertSerializationSame(string $functionName): void
     {
         $fixtureFile = $this->getFixtureFile($functionName);
-        $actualResult = $this->getGeneratorOutput();
+        $isXml = str_starts_with(basename($fixtureFile), 'Xml');
 
-        // file_put_contents( $fixtureFile, $actualResult );
-        // $this->markTestIncomplete( "Wrote fixture to '{$fixtureFile}'." );
+        $actualResult = $this->getGeneratorOutput();
 
-        self::assertSame(
-            file_get_contents($this->getFixtureFile($functionName)),
-            $actualResult
-        );
+        if ($isXml) {
+            $expectedXml = new \DOMDocument();
+            $expectedXml->preserveWhiteSpace = false;
+            $expectedXml->formatOutput = true;
+            $expectedXml->load($fixtureFile);
+
+            $actualXml = new \DOMDocument();
+            $actualXml->preserveWhiteSpace = false;
+            $actualXml->formatOutput = true;
+            $actualXml->loadXML($actualResult);
+
+            self::assertSame($expectedXml->saveXML(), $actualXml->saveXML());
```

Comment:
Have you tried to use `assertXmlStringEqualsXmlFile` instead?

Reply:
Yes and I think this method expects clear xml, without:
```xml
<?xml version="1.0" encoding="UTF-8"?>
```

because I'm getting:
> Could not read `<?xml version="1.0" encoding="UTF-8"?>`

with it.